test/unit/boa/class_methods_test.rb

```diff
@@ -33,13 +33,16 @@ class Properties < self
 
         sig { void }
         def test_properties
-          subject = Class.new(described_class)
+          PropCheck.forall(PropertyGenerators::Type.instance) do |type|
+            type    = T.let(type, Boa::Type)
+            subject = Class.new(described_class)
 
-          assert_empty(subject.properties)
+            assert_empty(subject.properties)
 
-          type = subject.properties[type_name] = Boa::Type::String.new(type_name)
+            subject.properties[type.name] = type
 
-          assert_equal({ name: type }, subject.properties)
```

Comment:
__TODO: Extract a precommit to change this line to:

```ruby
assert_equal({ type.name => type }, subject.properties)
```

Apply this change to other relevant places, preferring to use `type.name`, but `type_name` is ok if the `type` is not available. The `name:` key should not be hard coded in tests.